### DISCUSSION TITLE
Support for ETWS and CMAS messages.

### DISCUSSION THREAD

**Discussion by SAIRISAN123** [srsran/srsRAN_4G#1312] [General]
![317189432-fb49c72a-8cac-42e1-a326-164ea95333ce](https://github.com/srsran/srsRAN_4G/assets/118654165/9b1e73fc-740a-4023-8eea-4233c34ef805)
![317189438-c08891c7-e135-4300-8a7b-ad439a5b99ac](https://github.com/srsran/srsRAN_4G/assets/118654165/fe9a2d0e-2469-4dee-bf72-d2def6c80c8e)
![317189408-f9d0959b-e4da-4d9b-9037-4da922b5b494](https://github.com/srsran/srsRAN_4G/assets/118654165/f3cc791f-9772-452c-b2e5-90d303e0c33f)

Hi, @andrepuschmann  I have configured CBC and MME and can receive CMAS/ETWS messages from CBC to MME in open5gs. I am using OsmoCBC.  But the messages are not received by srsRAN eNB since there are no features for SIB 10,11,12. Can anyone tell me? what are the files I need to work on to send and receive CMAS with "write and replace warning" and their ACK? With the help of other repo implementations, I have already tried the PWS alert with sib. conf manually injecting messages and it is working fine.

**Comment by christianmair** [comment]
Did you figure out a working solution besides manually configuring SIB12 in the sib.conf (e.g. as shown in projects like [this](https://github.com/5G-MAG/rt-mbms-tx-for-qrd-and-crd/tree/emergency-alerts))?

**Reply by SAIRISAN123** [reply]
Not yet
